lightblue/src/main/java/org/esbtools/eventhandler/lightblue/model/DocumentEventEntity.java

```diff
@@ -61,13 +63,14 @@ public void setCanonicalType(String canonicalType) {
     }
 
     @Transient
-    public Optional<String> getParameterByKey(String key) {
+    @Nullable
+    public String getParameterByKey(String key) {
         for (KeyAndValue keyAndValue : parameters) {
             if (Objects.equals(key, keyAndValue.getKey())) {
-                return Optional.ofNullable(keyAndValue.getValue());
+                return keyAndValue.getValue();
             }
         }
-        return Optional.empty();
+        throw new NoSuchElementException(key);
```

Comment:
This change kinda snuck in here; model update just to allow for a `null` value as valid which could be meaningful versus not set at all.